File: frontend/src/types/download.ts
export type DownloadStatus =
    | 'queued'
    | 'downloading'
    | 'completed'
    | 'organizing'
    | 'organized'
    | 'failed';

export interface Download {
    id: string;
    title: string;
    author: string;
    series?: string;
    seriesNumber?: string;
    status: DownloadStatus;
    progress: number; // 0-100
    organized_path?: string;
    error_message?: string;
    created_at: string;
    completed_at?: string;
    organized_at?: string;
}

export interface CreateDownloadRequest {
    title: string;
    author: string;
    series?: string;
    seriesNumber?: string;
    category: string;
    torrent_url?: string;
    magnet_link?: string;
}

export interface BatchCreateDownloadRequest {
    downloads: CreateDownloadRequest[];
}

export interface BatchDownloadError {
    index: number;
    request: CreateDownloadRequest;
    error: string;
}

export interface BatchCreateDownloadResponse {
    successful: Download[];
    failed: BatchDownloadError[];
}

File: backend/internal/models/download.go
package models

import "time"

type Download struct {
	ID            string
	Title         string
	Author        string
	Series        string
	SeriesNumber  string
	TorrentURL    string
	MagnetLink    string
	TorrentBytes  []byte
	Category      string
	QBitHash      string
	Status        DownloadStatus
	Progress      float64
	DownloadPath  string
	OrganizedPath string
	ErrorMessage  string
	CreatedAt     time.Time
	CompletedAt   *time.Time
	OrganizedAt   *time.Time
}

type DownloadStatus string

const (
	StatusQueued      DownloadStatus = "queued"
	StatusDownloading DownloadStatus = "downloading"
	StatusCompleted   DownloadStatus = "completed"
	StatusOrganizing  DownloadStatus = "organizing"
	StatusOrganized   DownloadStatus = "organized"
	StatusFailed      DownloadStatus = "failed"
)
